library/log.go

```diff
@@ -36,6 +42,25 @@ func (l *Log) AppendData(data []byte) {
 	l.SetData(append(l.GetData(), data...))
 }
 
+// MaskData reads through the log data and masks
+// all values provided in the string slice. If the
+// log is empty, we do nothing.
+func (l *Log) MaskData(secrets []string) {
+	// convert data to string
+	strData := string(l.GetData())
+	for _, secret := range secrets {
+		secret = regexp.QuoteMeta(secret)
+		re := regexp.MustCompile((`(\s|^|"|:|'|\.|,)` + secret + `(\s|$|"|:|'|\.|,)`))
+		matches := re.FindAllString(strData, -1)
+		for _, match := range matches {
+			mask := string(match[0]) + constants.SecretLogMask + string(match[len(match)-1])
+			strData = strings.Replace(strData, match, mask, 1)
+		}
+		strData = re.ReplaceAllString(strData, constants.SecretLogMask)
+	}
```

Comment:
Looks like there is a lot going on here 😅 

Could we add comments to better illustrate what this code is doing?